src/datreant/core/collections.py

```diff
@@ -592,19 +594,23 @@ def __getitem__(self, index):
         # we can take lists of indices, names, or uuids; these return a
         # Bundle; repeats already not respected since Bundle functions as a
         # set
-        if ((isinstance(index, list) or hasattr(index, 'dtype')) and
-                all([isinstance(item, bool) for item in index])):
-            # boolean indexing
-            memberlist = self._list()
+        memberlist = self._list()
+
+        if ((isinstance(index, list) and all(isinstance(item, bool) for item in index))
+            or
+            (hasattr(index, 'dtype') and index.dtype == 'bool')):
```

Comment:
I would change the `hasattr` to be first. This will take advantage of lazy logic evaluation and might bring minor speed improvements for large bundles (gut feeling 1000 treants or more)